pkg/controller/directvolumemigration/rsync.go

```diff
@@ -553,6 +561,37 @@ func (t *Task) createRsyncTransferPods() error {
 	return nil
 }
 
+func getPodConfig(client k8sclient.Client, keys []string) (corev1.ResourceList, corev1.ResourceList, error) {
+	podConfigMap := &corev1.ConfigMap{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: "migration-controller", Namespace: migapi.OpenshiftMigrationNamespace}, podConfigMap)
+	if err != nil {
+		return nil, nil, err
+	}
+	limits := corev1.ResourceList{}
+	if _, exists := podConfigMap.Data[keys[0]]; exists {
```

Comment:
A slice of keys is passed in to this func, but then inside the func we're assuming that the first key is a CPU limit, the second is a memory limit, etc. This feels like a fragile way of handling it. Since this is used for both transfer and client pod, one possibility would be to just pass those strings in as individual function params -- replace `keys []string` with `cpu_limit, memory_limit, cpu_request, memory_request string` or something like that.

Reply:
+1 please take Scott's approach or pass in a struct instead

Reply:
+1

Reply:
Good catch! Updating code in regards to this